src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/BuildHostProcessManager.cs

```diff
@@ -33,22 +33,41 @@ public async Task<IBuildHost> GetBuildHostAsync(string projectFilePath, Cancella
     {
         var neededBuildHostKind = GetKindForProject(projectFilePath);
 
-        _logger?.LogTrace($"Choosing a build host of type {neededBuildHostKind} for {projectFilePath}");
+        _logger?.LogTrace($"Choosing a build host of type {neededBuildHostKind} for {projectFilePath}.");
 
+        var buildHost = await GetBuildHostAsync(neededBuildHostKind, cancellationToken).ConfigureAwait(false);
+
+        // If this is a .NET Framework build host, we may not have have build tools installed and thus can't actually use it to build.
+        // Check if this is the case.
+        if (neededBuildHostKind == BuildHostProcessKind.NetFramework)
+        {
+            if (!await buildHost.IsProjectFileSupportedAsync(projectFilePath, cancellationToken))
```

Comment:
This appears to be being called twice for every .net framework project (we call it here and then after this returns the build host).  I'm assuming that this fast enough that it shouldn't matter?

Or should we get the fallback build host in the caller, so that we only call this a second time if the first load failed in the .net framework host

Reply:
I guess I haven't benchmarked it; that said if you have multiple projects we'll be loading those in parallel which would be dominating your CPU time.

I didn't want to expose this to the caller since we'll need this logic in MSBuildWorkspace proper.